File: vol1/ch1/LeetCode27-3.cpp
class Solution {
public:
    int removeElement(vector<int>& nums,int val) {
        int j=-1;
        for(int i=0;i<nums.size();i++) {
            if (nums[i]!=val) {              //nums[i]为保留的元素
                j++;                         //扩大保留元素区间
                if (j!=i)                   //序号i和j的两个元素交换
                    swap(nums[i],nums[j]); 
            }
        }
        return j+1;                      //新数组长度为j+1
    }
};

File: vol1/ch1/LeetCode27-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        j=-1
        for i in range(0,len(nums)):
            if nums[i]!=val:              #nums[i]为保留的元素
                j+=1                         #扩大保留元素区间
                if j!=i:                   #序号i和j的两个元素交换
                    nums[i],nums[j]=nums[j],nums[i] 
        return j+1                     #新数组长度为j+1
